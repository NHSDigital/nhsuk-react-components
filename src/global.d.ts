declare module 'nhsuk-frontend/packages/components/header/header.js';
declare module 'nhsuk-frontend/packages/components/checkboxes/checkboxes.js';
declare module 'nhsuk-frontend/packages/components/character-count/character-count.js';
declare module 'nhsuk-frontend/packages/components/tabs/tabs.js';
