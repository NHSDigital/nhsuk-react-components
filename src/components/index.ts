export * from './content-presentation/index.js';
export * from './form-elements/index.js';
export * from './layout/index.js';
export * from './navigation/index.js';
export * from './typography/index.js';
export * from './utils/index.js';
