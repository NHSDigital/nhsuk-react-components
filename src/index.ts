export * from './components';
export * from './patterns';
