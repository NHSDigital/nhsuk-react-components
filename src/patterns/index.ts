export * from './nav-a-z/index.js';
export * from './review-date/index.js';
