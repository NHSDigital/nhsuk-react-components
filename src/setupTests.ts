import { TextDecoder, TextEncoder } from 'util';
import '@testing-library/jest-dom';
import { outdent } from 'outdent';

/**
 * Polyfill TextEncoder/TextDecoder for ReactDOM
 *
 * @see {@link https://github.com/jsdom/jsdom/issues/1695}
 */
Object.defineProperties(globalThis, {
  TextEncoder: {
    value: TextEncoder,
    configurable: true,
    enumerable: false,
    writable: true,
  },
  TextDecoder: {
    value: TextDecoder,
    configurable: true,
    enumerable: false,
    writable: true,
  },
});

/**
 * Polyfill Element methods for NHS.UK frontend error summary
 *
 * @see {@link https://github.com/jsdom/jsdom/issues/1695}
 */
Object.defineProperties(Element.prototype, {
  scrollIntoView: { value: jest.fn() },
});

/**
 * Polyfill `matchMedia()` for NHS.UK frontend tabs
 */
Object.defineProperty(globalThis, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: true,
    media: query,
    onchange: null,
    addListener: jest.fn(),
    addEventListener: jest.fn(),
    removeListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

beforeEach(() => {
  const stylesheet = document.createElement('style');

  stylesheet.innerHTML = outdent`
    :root {
      --nhsuk-breakpoint-mobile: 20rem;
      --nhsuk-breakpoint-tablet: 40.0625rem;
      --nhsuk-breakpoint-desktop: 48.0625rem;
      --nhsuk-breakpoint-large-desktop: 61.875rem;
    }
  `;

  // Add styles for NHS.UK frontend checks
  document.head.appendChild(stylesheet);

  // Flag NHS.UK frontend as supported
  document.body.classList.add('nhsuk-frontend-supported');
});
