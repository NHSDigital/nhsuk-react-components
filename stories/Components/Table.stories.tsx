import React from 'react';
import { Col, Hint, Row, Table } from '../../src';
import { Meta, StoryObj } from '@storybook/react';

const meta: Meta<typeof Table> = {
  title: 'Components/Table',
  component: Table,
};
export default meta;
type Story = StoryObj<typeof Table>;

Table.Head.displayName = 'Table.Head';
Table.Body.displayName = 'Table.Body';
Table.Row.displayName = 'Table.Row';
Table.Cell.displayName = 'Table.Cell';
Table.Panel.displayName = 'Table.Panel';

export const StandardTable: Story = {
  render: (args) => (
    <Table caption="Skin symptoms and possible causes">
      <Table.Head>
        <Table.Row>
          <Table.Cell>Skin Symptoms</Table.Cell>
          <Table.Cell>Possible cause</Table.Cell>
        </Table.Row>
      </Table.Head>
      <Table.Body>
        <Table.Row>
          <Table.Cell>Blisters on lips or around the mouth</Table.Cell>
          <Table.Cell>cold sores</Table.Cell>
        </Table.Row>
        <Table.Row>
          <Table.Cell>Itchy, dry, cracked, sore</Table.Cell>
          <Table.Cell>eczema</Table.Cell>
        </Table.Row>
        <Table.Row>
          <Table.Cell>Itchy blisters</Table.Cell>
          <Table.Cell>shingles, chickenpox</Table.Cell>
        </Table.Row>
      </Table.Body>
    </Table>
  ),
};

export const TablePanel: Story = {
  render: (args) => (
    <Table.Panel heading="Conditions similar to impetigo">
      <Table caption="Other possible causes of your symptoms">
        <Table.Head>
          <Table.Row>
            <Table.Cell>Skin Symptoms</Table.Cell>
            <Table.Cell>Possible cause</Table.Cell>
          </Table.Row>
        </Table.Head>
        <Table.Body>
          <Table.Row>
            <Table.Cell>Blisters on lips or around the mouth</Table.Cell>
            <Table.Cell>cold sores</Table.Cell>
          </Table.Row>
          <Table.Row>
            <Table.Cell>Itchy, dry, cracked, sore</Table.Cell>
            <Table.Cell>eczema</Table.Cell>
          </Table.Row>
          <Table.Row>
            <Table.Cell>Itchy blisters</Table.Cell>
            <Table.Cell>shingles, chickenpox</Table.Cell>
          </Table.Row>
        </Table.Body>
      </Table>
    </Table.Panel>
  ),
};

export const ResponsiveTable: Story = {
  args: { responsive: true },
  render: ({ responsive }) => (
    <Table responsive={responsive} caption="Ibuprofen syrup dosages for children">
      <Table.Head role="rowgroup">
        <Table.Row>
          <Table.Cell>Age</Table.Cell>
          <Table.Cell>How much?</Table.Cell>
          <Table.Cell>How often?</Table.Cell>
        </Table.Row>
      </Table.Head>
      <Table.Body>
        <Table.Row>
          <Table.Cell>3 to 5 months (weighing more than 5kg)</Table.Cell>
          <Table.Cell>2.5ml</Table.Cell>
          <Table.Cell>Max 3 times in 24 hours</Table.Cell>
        </Table.Row>
        <Table.Row>
          <Table.Cell>6 to 11 months</Table.Cell>
          <Table.Cell>2.5ml</Table.Cell>
          <Table.Cell>Max 3 to 4 times in 24 hours</Table.Cell>
        </Table.Row>
        <Table.Row>
          <Table.Cell>1 to 3 years</Table.Cell>
          <Table.Cell>5ml</Table.Cell>
          <Table.Cell>Max 3 times in 24 hours</Table.Cell>
        </Table.Row>
      </Table.Body>
    </Table>
  ),
};

export const NumericCells: Story = {
  render: (args) => (
    <Row>
      <Col width="one-half">
        <Hint>Right-aligned cells are used for numeric values</Hint>
        <Table caption="Number of cases">
          <Table.Head>
            <Table.Row>
              <Table.Cell>Location</Table.Cell>
              <Table.Cell isNumeric>Number of cases</Table.Cell>
            </Table.Row>
          </Table.Head>
          <Table.Body>
            <Table.Row>
              <Table.Cell>England</Table.Cell>
              <Table.Cell isNumeric>4,000</Table.Cell>
            </Table.Row>
            <Table.Row>
              <Table.Cell>Wales</Table.Cell>
              <Table.Cell isNumeric>2,500</Table.Cell>
            </Table.Row>
            <Table.Row>
              <Table.Cell>Scotland</Table.Cell>
              <Table.Cell isNumeric>600</Table.Cell>
            </Table.Row>
          </Table.Body>
        </Table>
      </Col>
    </Row>
  ),
};
