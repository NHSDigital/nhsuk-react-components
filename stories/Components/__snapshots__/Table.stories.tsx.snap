// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Table NumericCells smoke-test 1`] = `
<div class="nhsuk-grid-row">
  <div class="nhsuk-grid-column-one-half">
    <div class="nhsuk-hint">
      Right-aligned cells are used for numeric values
    </div>
    <table class="nhsuk-table">
      <caption class="nhsuk-table__caption">
        Number of cases
      </caption>
      <thead class="nhsuk-table__head">
        <tr class="nhsuk-table__row">
          <th class="nhsuk-table__header"
              scope="col"
          >
            Location
          </th>
          <th class="nhsuk-table__header nhsuk-table__header--numeric"
              scope="col"
          >
            Number of cases
          </th>
        </tr>
      </thead>
      <tbody class="nhsuk-table__body">
        <tr class="nhsuk-table__row">
          <td class="nhsuk-table__cell">
            England
          </td>
          <td class="nhsuk-table__cell nhsuk-table__cell--numeric">
            4,000
          </td>
        </tr>
        <tr class="nhsuk-table__row">
          <td class="nhsuk-table__cell">
            Wales
          </td>
          <td class="nhsuk-table__cell nhsuk-table__cell--numeric">
            2,500
          </td>
        </tr>
        <tr class="nhsuk-table__row">
          <td class="nhsuk-table__cell">
            Scotland
          </td>
          <td class="nhsuk-table__cell nhsuk-table__cell--numeric">
            600
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Components/Table ResponsiveTable smoke-test 1`] = `
<table class="nhsuk-table-responsive">
  <caption class="nhsuk-table__caption">
    Ibuprofen syrup dosages for children
  </caption>
  <thead class="nhsuk-table__head"
         role="rowgroup"
  >
    <tr class="nhsuk-table__row">
      <th class="nhsuk-table__header"
          scope="col"
      >
        Age
      </th>
      <th class="nhsuk-table__header"
          scope="col"
      >
        How much?
      </th>
      <th class="nhsuk-table__header"
          scope="col"
      >
        How often?
      </th>
    </tr>
  </thead>
  <tbody class="nhsuk-table__body">
    <tr class="nhsuk-table__row">
      <td class="nhsuk-table__cell"
          role="cell"
      >
        <span class="nhsuk-table-responsive__heading">
          Age
        </span>
        3 to 5 months (weighing more than 5kg)
      </td>
      <td class="nhsuk-table__cell"
          role="cell"
      >
        <span class="nhsuk-table-responsive__heading">
          How much?
        </span>
        2.5ml
      </td>
      <td class="nhsuk-table__cell"
          role="cell"
      >
        <span class="nhsuk-table-responsive__heading">
          How often?
        </span>
        Max 3 times in 24 hours
      </td>
    </tr>
    <tr class="nhsuk-table__row">
      <td class="nhsuk-table__cell"
          role="cell"
      >
        <span class="nhsuk-table-responsive__heading">
          Age
        </span>
        6 to 11 months
      </td>
      <td class="nhsuk-table__cell"
          role="cell"
      >
        <span class="nhsuk-table-responsive__heading">
          How much?
        </span>
        2.5ml
      </td>
      <td class="nhsuk-table__cell"
          role="cell"
      >
        <span class="nhsuk-table-responsive__heading">
          How often?
        </span>
        Max 3 to 4 times in 24 hours
      </td>
    </tr>
    <tr class="nhsuk-table__row">
      <td class="nhsuk-table__cell"
          role="cell"
      >
        <span class="nhsuk-table-responsive__heading">
          Age
        </span>
        1 to 3 years
      </td>
      <td class="nhsuk-table__cell"
          role="cell"
      >
        <span class="nhsuk-table-responsive__heading">
          How much?
        </span>
        5ml
      </td>
      <td class="nhsuk-table__cell"
          role="cell"
      >
        <span class="nhsuk-table-responsive__heading">
          How often?
        </span>
        Max 3 times in 24 hours
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Components/Table StandardTable smoke-test 1`] = `
<table class="nhsuk-table">
  <caption class="nhsuk-table__caption">
    Skin symptoms and possible causes
  </caption>
  <thead class="nhsuk-table__head">
    <tr class="nhsuk-table__row">
      <th class="nhsuk-table__header"
          scope="col"
      >
        Skin Symptoms
      </th>
      <th class="nhsuk-table__header"
          scope="col"
      >
        Possible cause
      </th>
    </tr>
  </thead>
  <tbody class="nhsuk-table__body">
    <tr class="nhsuk-table__row">
      <td class="nhsuk-table__cell">
        Blisters on lips or around the mouth
      </td>
      <td class="nhsuk-table__cell">
        cold sores
      </td>
    </tr>
    <tr class="nhsuk-table__row">
      <td class="nhsuk-table__cell">
        Itchy, dry, cracked, sore
      </td>
      <td class="nhsuk-table__cell">
        eczema
      </td>
    </tr>
    <tr class="nhsuk-table__row">
      <td class="nhsuk-table__cell">
        Itchy blisters
      </td>
      <td class="nhsuk-table__cell">
        shingles, chickenpox
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Components/Table TablePanel smoke-test 1`] = `
<div class="nhsuk-table__panel-with-heading-tab">
  <h3 class="nhsuk-table__heading-tab">
    Conditions similar to impetigo
  </h3>
  <table class="nhsuk-table">
    <caption class="nhsuk-table__caption">
      Other possible causes of your symptoms
    </caption>
    <thead class="nhsuk-table__head">
      <tr class="nhsuk-table__row">
        <th class="nhsuk-table__header"
            scope="col"
        >
          Skin Symptoms
        </th>
        <th class="nhsuk-table__header"
            scope="col"
        >
          Possible cause
        </th>
      </tr>
    </thead>
    <tbody class="nhsuk-table__body">
      <tr class="nhsuk-table__row">
        <td class="nhsuk-table__cell">
          Blisters on lips or around the mouth
        </td>
        <td class="nhsuk-table__cell">
          cold sores
        </td>
      </tr>
      <tr class="nhsuk-table__row">
        <td class="nhsuk-table__cell">
          Itchy, dry, cracked, sore
        </td>
        <td class="nhsuk-table__cell">
          eczema
        </td>
      </tr>
      <tr class="nhsuk-table__row">
        <td class="nhsuk-table__cell">
          Itchy blisters
        </td>
        <td class="nhsuk-table__cell">
          shingles, chickenpox
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
